src/modules/esl-utils/dom/script.ts

```diff
@@ -1,23 +1,22 @@
 /**
- * Creates new async script tag by id and src
+ * Creates new script element with specified attributes
  */
-const createAsyncScript = (id: string, src: string): HTMLScriptElement => {
+const createScript = (attrs: Record<string, string | boolean>): HTMLScriptElement => {
   const script = document.createElement('script');
-  script.id = id;
-  script.async = true;
-  script.src = src;
+  Object.entries(attrs).forEach(([name, value]) => value && script.setAttribute(name, value === true ? '' : value));
   return script;
 };
 
 /**
  * Common function that loads script async
  * @param id - unique script id that is used as a marker to prevent future load
  * @param src - script src (url) to load
+ * @param options - additional attributes to set on the script tag
```

Comment:
Mb lets limit the list of allowed parameters (at least from the types perspective but in that case i'd vote in general).
We are not providing utility to create script, the amount of capabilities should be strictly limited to the name an purpose of utility - load script file. So proposing to limit parmas list to cases we need in loading process :
 - crossorigin
 - referrerpolicy 
  - integrity (not sure we need)
 and probably that's it 
  We should not allow user to define useless attributes (defer, async, blocking, fetchprioriy, nonce, nomodeule - all of them will not work).
  `type` is also doesn't sound good here (we loading a simple old script, for loading modules user should consider browser capabilities to do that)

Reply:
Sounds fine but I believe we are not gonna support anything additional. 

And small note for ideal typing:
replace strings with exact literals / types where it makes sense

E.g.:
```
   crossorigin?: boolean | 'use-credentials' | 'anonimus'; // Note: true = 'anonimus'
```